cdap-formats/src/main/java/io/cdap/cdap/format/io/JsonStructuredRecordDatumWriter.java

```diff
@@ -122,6 +123,17 @@ protected void encodeUnion(Encoder encoder, Schema schema, int matchingIdx, Obje
     encode(encoder, schema.getUnionSchema(matchingIdx), value);
   }
 
+  /**
+   * Encode a BigDecimal instance
+   *
+   * @param encoder encoder
+   * @param decimal value to encode
+   * @throws IOException if the encoding did not succeed.
+   */
+  protected void encodeDecimal(Encoder encoder, BigDecimal decimal) throws IOException {
```

Comment:
It's much better to have it part of Encoder interface than doing this upcasting

Reply:
I didn't want to touch the encoder interface as it's also used elsewhere (KyroEncoder, for example)